client/index.js

```diff
@@ -65,10 +66,23 @@ async function doRender ({ Component, props, err }) {
     props = await loadGetInitialProps(Component, { err, pathname, query })
   }
 
+  // Try/catch is needed because IE11 has a CustomEvent implementation without contructor
+  try {
+    const event = new CustomEvent('before-reactdom-render', { detail: { Component } })
+    document.dispatchEvent(event)
+  } catch (e) {}
+
   Component = Component || lastAppProps.Component
   props = props || lastAppProps.props
 
   const appProps = { Component, props, err, router, headManager }
-  lastAppProps = appProps
   ReactDOM.render(createElement(App, appProps), container)
+
+  // Try/catch is needed because IE11 has a CustomEvent implementation without contructor
+  try {
+    const event = new CustomEvent('after-reactdom-render', { detail: { Component } })
```

Comment:
This looks not nice to me.
Shall we accept a new callback from the default function of this module and `next-dev` simply use it.

That's much better and clean.